src/itsdangerous/signer.py

```diff
@@ -159,14 +168,18 @@ def sign(self, value):
 
     def verify_signature(self, value, sig):
         """Verifies the signature for the given value."""
-        key = self.derive_key()
-
         try:
             sig = base64_decode(sig)
         except Exception:
             return False
 
-        return self.algorithm.verify_signature(key, value, sig)
+        for secret_key in self.secret_keys[::-1]:
```

Comment:
I think `reversed(self.secret_keys)` might be better now, Python 3 has special handling for reversed lists.